modin/core/dataframe/algebra/groupby.py

```diff
@@ -272,7 +272,10 @@ def reduce(
                 inplace=True,
             )
         # Result could not always be a frame, so wrapping it into DataFrame
-        return pandas.DataFrame(result)
+        result = pandas.DataFrame(result)
+        if result.index.name == MODIN_UNNAMED_SERIES_LABEL:
+            result.index.name = None
```

Comment:
Do we have a test that check `result.index.name`?

Reply:
Yes, two tests are failing if I remove this logic:
```
E           AssertionError: DataFrame.index are different
E
E           Attribute "names" are different
E           [left]:  ['__reduced__']
E           [right]: [None]

modin/pandas/test/utils.py:629: AssertionError
=========================== short test summary info ============================
FAILED modin/pandas/test/test_groupby.py::test_simple_row_groupby[False-True-by16] - AssertionError: DataFrame.index are different
FAILED modin/pandas/test/test_groupby.py::test_simple_row_groupby[True-True-by16] - AssertionError: DataFrame.index are different
```